### ISSUE TITLE
Runtime Error - SyntaxError: Unexpected token ')'  - Alpine Linux

### ISSUE THREAD

**Issue by cheynewallace** [vercel/next.js#33241]
### Run `next info` (available from version 12.0.8 and up)

    Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP Wed Aug 25 23:20:18 UTC 2021
    Binaries:
      Node: 16.13.1
      npm: 8.1.2
      Yarn: 1.22.15
      pnpm: N/A
    Relevant packages:
      next: 12.0.8
      react: 17.0.2
      react-dom: 17.0.2

### What version of Next.js are you using?

12.0.8

### What version of Node.js are you using?

16.13.1

### What browser are you using?

Chrome

### What operating system are you using?

Alpine Linux

### How are you deploying your application?

Docker Container Running node:16.13.1-alpine3.14 Image

### Describe the Bug

Im having an issue that I first saw after the release of NextJS 12 and im still seeing now with NextJS 12.0.8. It's completely stopping me from upgrading from version 11.

Essentially my application will build fine, no errors, and the server will start, but the second I make the first request to it, it trips over and I get the following highly unhelpful error

```
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1031:15)
    at Module._compile (node:internal/modules/cjs/loader:1065:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.__webpack_require__.f.require (/app/.next/server/webpack-runtime.js:143:28)
    at /app/.next/server/webpack-runtime.js:70:40
/app/.next/server/chunks/6859.js:135
/***/ }),
       ^

SyntaxError: Unexpected token ')'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1031:15)
    at Module._compile (node:internal/modules/cjs/loader:1065:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.__webpack_require__.f.require (/app/.next/server/webpack-runtime.js:143:28)
    at /app/.next/server/webpack-runtime.js:70:40
```

The problem seems to be running on Alpine linux as I can perform a production build locally, then run it with no issues However after building my production docker container and deploying it, I see a white screen, with this error in the console.

Googling this has come up with nothing, checking the source at that point reveals nothing interesting to me.

My app has been working fine on Next JS 11, but upgrading to 12 blows up. I see nothing in the docs about needing to do anything different with regards to docker or alpine linux.

Can any one help?

### Expected Behavior

The app should run, like it did with NextJS 11

### To Reproduce

Upgrade from Next 11 to Next 12 in a container running `node:16.13.1-alpine3.14` perhaps? Im not sure how to reproduce it out of our environment. All we did was bump up the Next version in package.json then yarn install.  

Im unsure if I need to be installing additional packages in the container?

**Comment by balazsorban44** [comment]
Hi, please add a reproduction so we can investigate. Ideally, a repository with some code that reproduces this issue, and probably your Dockerfile would be useful as well.

We have a whole section on how to properly host through docker that might be of interest:

https://nextjs.org/docs/deployment#docker-image
https://github.com/vercel/next.js/tree/canary/examples/with-docker

**Comment by cheynewallace** [comment]
I can't share our application code base as it is private software, but our application container is based on the Dockerfile in the link you posted, we've been running in production without issue since NextJS 10.     Im going to assume simply running a brand new NextJS 12 application wont show the problem (as im sure that would of come up) so wondering if there were any other migration steps, like additional packages that needed to be installed. 

Is there anything in particular that needed to change for NextJS 12 with regards to containers or linux packages?  There does not seem to be any information about that.    I have a feeling it may be to do with SWC on Alpine?  For instance https://github.com/swc-project/swc/issues/1129

**Comment by cheynewallace** [comment]
For reference here's our dockerfile

```# Install dependencies only when needed
FROM node:16.13.1-alpine3.14 AS deps
# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
RUN apk add --no-cache libc6-compat
WORKDIR /app
COPY package.json yarn.lock ./
RUN yarn install --frozen-lockfile

# Rebuild the source code only when needed
FROM node:16.13.1-alpine3.14 AS builder
WORKDIR /app
COPY --from=deps /app/node_modules ./node_modules
COPY . .
RUN source .env.production.local
RUN yarn build
RUN /bin/sh scripts/clean-maps.sh

# Production image, copy all the files and run next
FROM node:16.13.1-alpine3.14 AS runner
WORKDIR /app

ENV NODE_ENV production

RUN addgroup -g 1001 -S nodejs
RUN adduser -S nodejs -u 1001

COPY --from=builder /app/next.config.js ./
COPY --from=builder /app/public ./public
COPY --from=builder --chown=nodejs:nodejs /app/.next ./.next
COPY --from=builder /app/node_modules ./node_modules
COPY --from=builder /app/package.json ./package.json
COPY --from=builder /app/release-version ./release-version
COPY  package.json  \
      yarn.lock \
      sentry.properties \
      sentry.client.config.js \
      sentry.server.config.js \
      ./

USER nodejs

EXPOSE 3000

ARG VERSION
ENV SENTRY_RELEASE=$VERSION
ENV NEXT_TELEMETRY_DISABLED 1

CMD ["yarn", "start"]```

**Comment by balazsorban44** [comment]
I wouldn't need the whole codebase, just a reproduction. Start with your project, and try removing parts that don't make a difference, until you are left with something that could be shared and still reprodocues the issue.

**Comment by cheynewallace** [comment]
Fair enough, I realize its a pretty unhelpful error to work with, was hoping it had maybe been seen before.
I'll see if I can come up with something.

**Comment by cheynewallace** [comment]
So i'll close this one out, I discovered what the issue was and it was not to do with Alpine linux. 

We have a process that clears out any mention of source maps along with a few other cleanup steps.     
The script we've been using lo locate and sanitize looks for mentions of `//# sourceMappingURL=head-manager.js.map` in the outputted JS files. 

It would seem that in the latest versions of Next the SourceMap declaration is on the same line as the closing bracket, where as previously it was on a new line.   The script was removing the line which in turn broke the closing bracket.

So, in next 12 it looks like 
```
    };
} //# sourceMappingURL=head-manager.js.map
```

in next 11 it was 

```
    };
} 
//# sourceMappingURL=head-manager.js.map
```

We've fixed this up and its now building properly.    Our fault, but thought it was worth reporting just incase.
